front/src/modules/powerRestriction/hooks/usePowerRestrictionSelectorBehaviours.ts

```diff
@@ -142,8 +153,19 @@ const usePowerRestrictionSelectorBehaviours = ({
     }
 
     // clean cut positions
-    setCutPositions((prev) =>
-      prev.filter((position) => position <= fromPosition || newToPosition <= position)
+    setCustomRanges((prev) =>
+      prev.reduce((acc, range, index) => {
+        if (index === selectedIntervalIndex) {
+          acc.push({ ...range, end: newToPosition });
+          return acc;
+        }
+        if (index === selectedIntervalIndex + 1) {
+          return acc;
+        }
```

Comment:
```suggestion
        if (range.begin === fromPosition) {
          acc.push({ ...range, end: newToPosition });
          return acc;
        }
        if (range.end === newToPosition) {
          return acc;
        }
```
I think we should compare the positions here since the `selectedIntervalIndex` is based on the `data` array we receive

Reply:
Done ;) I added some comments too